File: Python/D7 - W/5/5.py
# Write a python program to sum of digits of the number and check number is armstrong or not.

def armstrong_num(b):
    sum = 0
    while(b != 0):
        digit = b % 10
        sum = sum + (digit ** 3)
        b = b // 10
    return sum

num = int(input('Enter the number : '))
is_armstrong = armstrong_num(num)

if(num == is_armstrong):
    print("Number is armstrong number.")
else:
    print("Number is not armstrong number.")   

File: C/D11/5 - armstrong num/5.cpp
// Write a C program to check given number is armstrong number or not using function
// armstrong number ---> sum of cube of every digit in number = entered number
// example :153

#include <stdio.h>
#include <math.h>
void armstrong_num();

int main(){
    armstrong_num();
}

void armstrong_num(){
    int num, entered_num, digit = 0, sum = 0;

    printf("Enter number : ");
    scanf("%d",&num);

    entered_num = num;  

    while(num != 0){
        digit = num % 10;
        sum = sum + pow(digit, 3);
        num = num / 10;
    }

    entered_num == sum ? printf("Number is armstrong number.") : printf("Number is not armstrong number.");
}